platforms/arm.go

```diff
@@ -0,0 +1,69 @@
+package platforms
+
+import (
+	"bufio"
+	"log"
+	"os"
+	"runtime"
+	"strings"
+)
+
+// Structure to identify ARM platforms
+type PlatformArm struct {
+	// Present the ARM instruction set architecture, eg: v7, v8
+	Variant string
+	// Present the features that could not be detectable through HWCAPS
+	Features []string
+}
+
+func isArmArch(arch string) bool {
+	switch arch {
+	case "arm", "armbe", "arm64", "arm64be":
+		return true
+	}
+	return false
+}
+
+func isPattern(str string, pattern string) bool {
+	// The pattern in /proc/cpuinfo ends with \t e.g Features\t
+	return strings.EqualFold(strings.Trim(str, "\t"), pattern)
+}
+
+// For Linux, the kernel has already detected the ABI, ISA and Features.
+// So we don't need to access the ARM registers to detect platform information
+// by ourselves. We can just parse these information from /proc/cpuinfo
+func ParseCpuInfo(pattern string) string {
```

Comment:
- Does this need to be exported function?
- This should return err (or just "unknown") rather than call log.Fatal()
- ParseCpuInfo → getCPUVariant?
- Please add tests

Reply:
This is definitely not a package export.

This should also be in a separate file, probably `cpuinfo.go`.

Reply:
Hi Akihiro,

>Does this need to be exported function?

No, I think it's enough to used this function inside this package.

>This should return err (or just "unknown") rather than call log.Fatal()

Your comment is reasonable, I will refine this function to return proper "variant, err"

>ParseCpuInfo → getCPUVariant?

I think it's better to rename ParseCpuInfo to getCPUInfo. Because this function
is not only used to parse Variant. We may parse other information like "features"
in the future. But I will add a new function getCPUVariant to wrap getCPUInfo.

>Please add tests

Sorry, I am new to here. Did you mean something like platform_test.go?

Hi Stevvooe,

>This is definitely not a package export.
This should also be in a separate file, probably cpuinfo.go.
Did you mean rename the arm.go to cpuinfo.go? Because this function has been in
a separate file already.
 
Thanks for your comments. I am new to golang :) 

Add @Pennyzct who is working with me on arm-multi-arch image support.